cctrl/common.py

```diff
@@ -72,45 +73,31 @@ def init_api(settings):
                      encode_email=settings.encode_email)
 
 
-def get_email_and_password(settings):
-    # check ENV for credentials first
-    try:
-        email = os.environ.pop(settings.login_creds['email'])
-    except KeyError:
-        email = get_email(settings)
-
-    try:
-        password = os.environ.pop(settings.login_creds['pwd'])
-    except KeyError:
-        password = get_password()
-
-    return email, password
+def execute_command(api, command, settings):
+    while True:
+        try:
+            command()
+            break
+        except (cclib.TokenRequiredError, cclib.UnauthorizedError):
+            email, password = get_email_and_password(settings)
+            create_token(api, settings, email, password)
+        except ParseAppDeploymentName:
+            sys.exit(messages['InvalidAppOrDeploymentName'])
 
 
 def execute_with_authenticated_user(api, command, settings):
     while True:
         try:
-            try:
-                command()
-            except (cclib.TokenRequiredError, cclib.UnauthorizedError):
-                email, password = get_email_and_password(settings)
-                try:
-                    api.create_token(email, password)
-                except cclib.UnauthorizedError:
-                    sys.exit(messages['NotAuthorized'])
-                else:
-                    pass
-            except ParseAppDeploymentName:
-                sys.exit(messages['InvalidAppOrDeploymentName'])
-            else:
-                break
+            execute_command(api, command, settings)
+            break
         except cclib.ForbiddenError, e:
             sys.exit(messages['NotAllowed'])
         except cclib.ConnectionException:
             sys.exit(messages['APIUnreachable'])
-        except (cclib.BadRequestError, cclib.ConflictDuplicateError, cclib.GoneError,
-                cclib.InternalServerError, cclib.NotImplementedError, cclib.ThrottledError,
-                InputErrorException), e:
+        except (cclib.BadRequestError, cclib.ConflictDuplicateError,
+                cclib.GoneError, cclib.InternalServerError,
+                cclib.NotImplementedError, cclib.ThrottledError,
+                cclib.NotFoundError, InputErrorException), e:
```

Comment:
just to clarify, all exceptions descending from `cclib.APIException` will be excepted here?